src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,307 @@
+        wrongPhoneNumbers.addAll(getStringsWithSpecialSymbol("+38.050.223", "32.23", correctCharacters));
+        wrongPhoneNumbers.addAll(getStringsWithSpecialSymbol("+38-050-223-", "2-23", correctCharacters));
+        wrongPhoneNumbers.addAll(getStringsWithSpecialSymbol("+38.050.223.", "2.23", correctCharacters));
+
+        return wrongPhoneNumbers;
+    }
+
+    private static List<String> createCorrectUserNames() {
+        List<String> correctNames = new ArrayList<>(List.of(
+                "pushkin",
+                "ThisNameHasLengthMoreThan100SymbolsThisNameHasLengthMoreThan100SymbolsThisNameHasLengthMoreThan100Symbols",
+                "~~``~~",
+                "`_'",
+                "'@'\"`!",
+                "1",
+                "$a",
+                "$",
+                "a$",
+                "$+a",
+                "a+$"));
+
+        correctNames.addAll(getStringsWithSpecialSymbol("pu", "shkin", Collections.emptyList()));
+        correctNames.addAll(getStringsWithRussianOrUkrCharacter("aaa", "bbb"));
+
+        return correctNames;
+    }
+
+    private static List<String> createWrongUserNames() {
+
+        return List.of(
+                "",
+                " ",
+                "aaa bbb",
+                " aaa",
+                "aaa ");
+    }
+
+    private static List<String> createCorrectWordsEmptyOrMin2Max50() {
+        List<String> correctWords = new ArrayList<>(List.of(
+                "",
+                "aA",
+                "11",
+                "aA1",
+                "aA1-",
+                "aA1-a",
+                "aA1'",
+                "aA1'a",
+                "aA1`",
+                "aA1`a",
+                "aA1_",
+                "aA1_a",
+                "LengthIs50LengthIs50LengthIs50LengthIs50LengthIs50"));
+
+        correctWords.addAll(getStringsWithRussianOrUkrCharacter("aa", "bb"));
+
+        return correctWords;
+    }
+
+    private static List<String> createWrongWordsEmptyOrMin2Max50() {
+        List<String> wrongWords = new ArrayList<>(List.of(
+                "w",
+                "W",
+                "1",
+                " ",
+                " 1",
+                " W",
+                " w",
+                "1 ",
+                "1 ",
+                "W ",
+                "w ",
+                "ThisWordHasLengthMoreThan50CharactersItsLengthIs51S"));
+
+        List<Character> correctCharacters = List.of('\'', '_', '`', '-');
+        wrongWords.addAll(getStringsWithSpecialSymbol("aaa", "bbb", correctCharacters));
+        wrongWords.addAll(getStringsWithSpecialSymbol("", "", Collections.emptyList()));
+        wrongWords.addAll(getStringsWithRussianOrUkrCharacter("", ""));
+
+        return wrongWords;
+    }
+
+    private static List<String> getStringsWithSpecialSymbol(String str1, String str2, List<Character> deletedCharacters) {
+        List<Character> addedCharacters = new ArrayList<>(List.of(
```

Comment:
Nice job, but such code quite difficult to service. You can do it much effective using ASCII keys (example http://www.asciitable.com). And to collect all characters with IntRange:
```
List<Character> addedCharacters = new ArrayList<>();
IntStream.range(33, 48).forEach(i -> addedCharacters.add((char) i));
IntStream.range(58, 65).forEach(i -> addedCharacters.add((char) i));
IntStream.range(91, 97).forEach(i -> addedCharacters.add((char) i));
```
But you can do it more clear.

Reply:
Thanks, I think about it